payjoin/src/receive/multi_party/mod.rs

```diff
@@ -0,0 +1,253 @@
+use bitcoin::{FeeRate, Psbt};
+use error::{InternalMultiPartyError, MultiPartyError};
+
+use super::error::InputContributionError;
+use super::{v1, v2, Error, ImplementationError, InputPair};
+use crate::psbt::merge::merge_unsigned_tx;
+use crate::receive::v2::SessionContext;
+
+pub(crate) mod error;
+
+const SUPPORTED_VERSIONS: &[usize] = &[2];
+
+#[derive(Default)]
+pub struct UncheckedProposalBuilder {
+    proposals: Vec<v2::UncheckedProposal>,
+}
+
+impl UncheckedProposalBuilder {
+    pub fn new() -> Self { Self { proposals: vec![] } }
+
+    pub fn try_add(
```

Comment:
`try_` prefix is for when you have an infallible option in contrast. This should just be called `add`

Reply:
done